### PR TITLE
Add timezone to SiP time

### PR DESCRIPTION
**Pull request by Mottie**
## Description

In a recent staging review, the save-in-progress (SiP) title should include a comma after the date and include a time zone label

<details><summary>Current <code>SaveInProgressIntro</code> content</summary>

save-in-progress card showing date without a separating command or timezone indicator</details>

Two problems were encountered:
1) the `moment.js` timezone library was removed due to excessive size
2) I discovered that the [`date-fns-tz` library provides three-letter time zone (e.g. `EST`) "short" abbreviations](https://github.com/marnusw/date-fns-tz#format) which doesn't follow the [design system timezone abbreviation recommendation](https://design.va.gov/content-style-guide/dates-and-numbers#times-and-time-zones) of two-letter abbreviations (e.g. `ET`)

I discussed this problem with [Randi & Beth in Slack](https://dsva.slack.com/archives/C01K37HRUAH/p1641915123035600) where it was decided to show the library's provided three-letter timezone abbreviation.

Note: I also updated:
- `SaveStatus` - internal form (green background) SiP message
- `FormSaved` - info alert shown after using "Finish this application later" 
- `SaveInProgressIntro` - component with "Continue your application" and "Start a new application" buttons 
- `ApplicationStatus` component, but I believe this isn't being used any more - see https://dsva.slack.com/archives/C52CL1PKQ/p1641923817153100

## Original issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/34810

## Testing done

Updated unit tests

## Screenshots

<details><summary><code>SaveStatus</code> (with timezone)</summary>

Form changed save message</details>

<details><summary><code>FormSaved</code> (with timezone)</summary>

application has been saved message</details>

<details><summary><code>SaveInProgressIntro</code> (with timezone)</summary>

continue application message upon return</details>

## Acceptance criteria
- [x] `SaveStatus` updated with timezone
- [x] `FormSaved` updated with timezone
- [x] `SaveInProgressIntro` updated with timezone
- [x] `ApplicationStatus` updated with timezone
- [x] All tests passing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by bkjohnson — COMMENTED** [review]
This seems like a bit of a gray area. [Current guidance](https://design.va.gov/content-style-guide/dates-and-numbers#times-and-time-zones) says:

> Always use the time zone ET, even if the office is in another location

To me that's talking about dates/times on static pages for VAMCs, but "Always use the time zone ET" is what gets me. If we're going to be presenting times relative to the user's location (regardless of what page they're on) I think we may want to update the guidance. @k80bowman do you have any thoughts?

image

**Comment by Mottie** [comment]
I did [discuss this in Slack with Beth Potts](https://dsva.slack.com/archives/C01K37HRUAH/p1641921856038100?thread_ts=1641915123.035600&cid=C01K37HRUAH) and this was her response:

> Our style guide entry for time zones was written for the use case of hours of operation for VA offices. I think we should add to that entry for this use case of an individual's actions--whether saving in progress work or submitting an application or request. I do think it would be kind of weird and possibly confusing to show a West Coast user ET for when they saved their work in progress. So let's go with the time zone abbreviations in the open source date library and show the user's local time.

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 1]
```diff
@@ -92,15 +93,18 @@ class SaveInProgressIntro extends React.Component {
         const lastUpdated =
           savedForm.lastUpdated || _.get('metadata.lastUpdated', savedForm);
         const savedAt = this.props.lastSavedDate
-          ? moment(this.props.lastSavedDate)
-          : moment.unix(lastUpdated);
-        const expiresAt = moment.unix(savedForm.metadata.expiresAt);
-        const expirationDate = expiresAt.format('MMMM D, YYYY');
-        const isExpired = expiresAt.isBefore();
+          ? new Date(this.props.lastSavedDate)
+          : fromUnixTime(lastUpdated);
```
Comment (Mottie) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:97]: Cypress tests are failing because `lastUpdated` is undefined and `date-fns` is not as tolerant as moment.js. I'll look at refactoring this tomorrow.

**Comment by Mottie** [comment]
Updated multiple forms e2e tests because switching to use `date-fns` so the application status component title can include a timezone.

**Review by alexandec — APPROVED** [review]

**Review by devonjhills — APPROVED** [review]

**Review by craigwheeler — APPROVED** [review]
LGTM! 🚀

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 2]
```diff
@@ -89,25 +89,43 @@ class SaveInProgressIntro extends React.Component {
...
+         * savedForms.metadata.expiresAt = unix time
+         * savedForms.metadata.lastUpdated = unix time
+         * savedForms.metadata.savedAt = JS time (ms)
+         */
+        const { metadata = {} } = savedForm;
+        const lastUpdated = savedForm.lastUpdated || metadata.lastUpdated;
+
+        let savedAt = '';
+        if (this.props.lastSavedDate) {
+          savedAt = new Date(this.props.lastSavedDate);
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:104]: Good catch - if `lastSavedDate` is never passed in as a prop maybe we can remove it from this forms component @micahchiang . This would be in another PR.

image

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 3]
```diff
@@ -89,25 +89,43 @@ class SaveInProgressIntro extends React.Component {
...
+            )}`;
+
           const H = `h${this.props.headingLevel}`;
           includesFormControls = true;
           alert = (
             <div className="usa-alert usa-alert-info background-color-only schemaform-sip-alert">
               <div className="schemaform-sip-alert-title">
                 <H className="usa-alert-heading vads-u-font-size--h3">
-                  {inProgressMessage} and was last saved on {lastSavedDateTime}
+                  {inProgressMessage} {lastSavedDateTime}
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:128]: I think I prefer `lastSavedDateTime` just being the date as opposed to also having "and was last saved on" in there.

**Review by bkjohnson — COMMENTED** [review]
Can you link the branch/PR for updating the guidance?

File: src/platform/forms/save-in-progress/FormSaved.jsx [hunk 4]
```diff
@@ -42,9 +43,9 @@ class FormSaved extends React.Component {
       profile && profile.prefillsAvailable.includes(formId)
     );
     const { success } = this.props.route.formConfig.savedFormMessages || {};
-    const expirationDate = moment
-      .unix(this.props.expirationDate)
-      .format('MMMM D, YYYY');
+    const expirationDate = this.props.expirationDate
+      ? format(fromUnixTime(this.props.expirationDate), 'MMMM d, yyyy')
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/platform/forms/save-in-progress/FormSaved.jsx:47]: The expiration date is only showing the date, not the time. See the `FormSaved` and `SaveInProgressIntro` screenshots in the first comment.

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx (hunk 3 above)
Comment (Mottie) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:128]: I changed it because some Cypress tests don't include a last saved time and as such that message would be "Your application is in progress and was last saved on" with no date. I don't know if this would happen with production data, but it'd be less confusing if it did.

Would you rather have me add that last part conditionally then?

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx (hunk 2 above)
Comment (Mottie) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:104]: I feel like it was originally meant to be used for unit tests, but wasn't.

**Comment by Mottie** [comment]
> Can you link the branch/PR for updating the guidance?

The guidance hasn't been done yet. Here is the related ticket: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/35445

**Comment by k80bowman** [comment]
> To me that's talking about dates/times on static pages for VAMCs, but "Always use the time zone ET" is what gets me. If we're going to be presenting times relative to the user's location (regardless of what page they're on) I think we may want to update the guidance. @k80bowman do you have any thoughts?

Sorry I'm late in responding to this. This does feel like a place where we definitely need to update our guidance. It doesn't make sense for SiP time to be reported in eastern time if the user is in a different timezone. And it seems that having the two-letter timezone indicator is difficult from a technical perspective. @Mottie I see where you said Randi is going to update this, let me know if any assistance from our team is needed.

**Comment by bkjohnson** [comment]
@Mottie or @RLHecht do you know when the guidance will be updated? Ideally this PR would go live after that happens, but if it will take a little while I don't want to block this change - it looks fine.

**Comment by RLHecht** [comment]
@bkjohnson It will probably be a few more weeks. We don't have this slated for this sprint but could probably add it next sprint.

**Review by dan-brooking — APPROVED** [review]

**Comment by Mottie** [comment]
@bkjohnson Are there any other concerns you have about this PR?

**Review by bkjohnson — APPROVED** [review]
Nope, looks good :+1:
